ddtrace/contrib/vertica/patch.py

```diff
@@ -0,0 +1,265 @@
+        app_type=AppTypes.db,
+        tags=tags,
+        _config=config.vertica["patch"]["vertica_python.vertica.cursor.Cursor"],
+    )
+    pin.onto(cursor)
+
+
+# tracing configuration
+config._add(
+    "vertica",
+    {
+        "service_name": "vertica",
+        "app": "vertica",
+        "app_type": "db",
+        "patch": {
+            "vertica_python.vertica.connection.Connection": {
+                "routines": {
+                    "cursor": {"trace_enabled": False, "span_end": cursor_after}
+                }
+            },
+            "vertica_python.vertica.cursor.Cursor": {
+                "routines": {
+                    "execute": {
+                        "operation_name": "vertica.query",
+                        "span_type": sql.TYPE,
+                        "span_start": execute_span_start,
+                        "span_end": execute_span_end,
+                    },
+                    "copy": {
+                        "operation_name": "vertica.copy",
+                        "span_type": sql.TYPE,
+                        "span_start": copy_span_start,
+                    },
+                    "fetchone": {
+                        "operation_name": "vertica.fetchone",
+                        "span_type": "vertica",
+                        "span_end": fetch_span_end,
+                    },
+                    "fetchall": {
+                        "operation_name": "vertica.fetchall",
+                        "span_type": "vertica",
+                        "span_end": fetch_span_end,
+                    },
+                    "nextset": {
+                        "operation_name": "vertica.nextset",
+                        "span_type": "vertica",
+                        "span_end": fetch_span_end,
+                    },
+                }
+            },
+        },
+    },
+)
+
+
+def patch():
+    global _PATCHED
+    if _PATCHED:
+        return
+
+    _install(config.vertica)
+    _PATCHED = True
+
+
+def unpatch():
+    global _PATCHED
+    if _PATCHED:
+        _uninstall(config.vertica)
+        _PATCHED = False
+
+
+def _uninstall(config):
+    for patch_class_path in config["patch"]:
+        patch_mod, _, patch_class = patch_class_path.rpartition(".")
+
+        for patch_routine in config["patch"][patch_class_path]["routines"]:
+            mod = importlib.import_module(patch_mod)
+            cls = getattr(mod, patch_class)
+            unwrap(cls, patch_routine)
+
+
+def _find_routine_config(config, instance, routine_name):
+    """Attempts to find the config for a routine based on the bases of the
+    class of the instance.
+    """
+    bases = instance.__class__.__mro__
+    for base in bases:
+        full_name = "{}.{}".format(base.__module__, base.__name__)
+        if full_name not in config["patch"]:
+            continue
+
+        config_routines = config["patch"][full_name]["routines"]
+
+        if full_name in config["patch"] and routine_name in config_routines:
```

Comment:
`s/full_name in config["patch"] and //`

We are already doing that assertion above.